compose/cli/main.py

```diff
@@ -720,6 +724,9 @@ def up(self, project, options):
             raise UserError("--abort-on-container-exit and -d cannot be combined.")
 
         with up_shutdown_context(project, service_names, timeout, detached):
+            # start the event stream first so we don't lose any events
+            event_stream = project.events()
```

Comment:
Or maybe this case isn't a big concern, because we already attach to existing containers, it's very unlikely that something is going to scale a container before we start the event stream.